s30384-pj-AleksanderKoss/zadania3-28.10.23.txt

```diff
@@ -0,0 +1,133 @@
+ZADANIE 0 
+- Przechodzę do katalogu domowego:
+
+cd ~
+
+ls -a
+
+- Wypisało mi kilka plików i katalogów
+
+ls -a | grep k 
+
+- Wypisało mi tylko pliki i katalogi zawierające literę "k"
+
+ls -a | grep k > lista_plikow_z_literka_k.txt
+
+- utworzył się plik "lista_plikow_z_literka_k.txt"
+
+cat lista_plikow_z_literka_k.txt
+
+- jest w nim taka sam treść jak nazwa pliku "lista_plikow_z_literka_k.txt"
+
+| - to operator tzw. potok, który służy do przekierowania wynik wykonania 
+pierwszego polecenia do kolejnego polecenia, w przypadku ls -a | grep k 
+przekierowuje wynik ls-a do polecenia grep k, czyli wylistowuje pliki ale
+z przekierowaniem żeby wyrzucało tylko pliki zawierające "k"
+
+> - to operator przekierowania do pliku z nadpisaniem jego zawartości 
+
+ls -a | grep D | tee lista_plikow_z_literka_k_2.txt
+
+treść drugiego pliku zawiera w sobie również treść pierwszego pliku z literką "d".
+tee jest używana do przekazywania strumieni danych, a jednocześnie zapisywania 
+ich do pliku lub kilku plików. 
+
+ZADANIE 1
+
+ps 
+
+- wyświetla procesy związane z aktywna sesja terminala
+
+ps -a
+
+- wyświetla procesy oprócz lidera sesji 
+
+ps x
+
+- wyświetla procesy niezwiązane z aktualna sesją terminala 
+
+ps axu
+
+- wyświetla wszystkie procesy w systemie, wraz z informacjami 
+- znak ? iznacza, że proces nie jest związany z żadnym konkretnym terminalem. 
+To może dotyczyć na przykład procesów demonów, które działają w tle, 
+niezależnie od konkretnej sesji terminala.
+
+Wyświetl wszystkie procesy bash:
+
+ps axu | grep bash 
+
+- wyświetlam wszystkie procesy systemu a następnie przekierowuje wynik zeby wyrzucal tylko te linie, które zawierają "bash"
```

Comment:
No właśnie. Jeśli będziemy mieć usera o loginie `barabash` a ten będzie miał uruchomiony program o nazwie `zbrodniarz` to też zostanie wypisany pomimo, że jego proces nie nazywa się bash. Jak to skorygować? Jakiego parametru użyć?